app/assets/javascripts/times.js

```diff
@@ -0,0 +1,130 @@
+/* global $ */
+
+/* Datepicker */
+$(document).on('turbolinks:load', 
+    function (){
+        $('.datepicker').datepicker({todayHighlight: true, multidate: true, format: 'yyyy-mm-dd'});
+    }
+);
+
+/* Show highlighted datepicker dates */
+$(document).on('turbolinks:load',
+    function(){
+        $('.datepicker').datepicker().on('changeDate', function(e){
+            var string = $('.datepicker').val();
+            var dateArray = string.split(",");
+            var formatted = "";
+            
+            for (var i = 0; i < dateArray.length && string.length > 0; i++){
+                var unformatted = dateArray[i];
+                var niceFormat = formatDate(unformatted);
+                var element = "#"+unformatted;
+                if ($(element).length == 0){
+                    addDiv(unformatted);
+                }
+                formatted += niceFormat + '\n';
+            }
+            
+            deleteDiv(dateArray);
+            $(".dates-chosen").text(formatted);
+        });
+    }
+);
+
+/* Add Button functionality */
+$(document).on('click', '.addbutton', function(){
+    var parent_id = $(this).parent().attr('id');
+    var element = document.getElementById(parent_id);
+    var strid = "#"+parent_id;
+    var entry = createTimeEntry(parent_id);
+    element.append(entry);
+    console.log($(strid).children());
+    console.log($(strid).children().last());
+    $(strid).children().last().append('<span><button type="button" class="deletebutton btn btn-danger btn-sm">Delete Timeslot</button></span>');
+});
+
+/* Delete Button functionality */
+$(document).on('click', '.deletebutton', function(){
+    var parent = $(this).parent().parent();
+    parent.remove();
+});
+
+
+/* Time Entry Row */
+function createTimeEntry(divId){
+    var startInput = document.createElement("input");
+    startInput.type = "time";
+    startInput.value = "12:00";
+    startInput.step = "900";
+    startInput.name = "times["+divId +"][]";
+    var endInput = document.createElement("input");
+    endInput.type = "time";
+    endInput.value = "13:00";
+    endInput.step = "900";
+    endInput.name = "times["+divId +"][]";
+    
+    var outerDiv = document.createElement("div");
+    var startTime = document.createElement("span");
+    var endTime = document.createElement("span");
+    var startText = document.createTextNode("Start Time ");
+    var endText = document.createTextNode("End Time ");
+    startTime.appendChild(startText);
```

Comment:
Need to create some checks:
- End time > Start time
- Times that cross midnight, perhaps cap at midnight (maybe don't need for now).
- Suggestion:  When start time is moved forward, move end time forward by end_time - start_time